test/e2e_env/universal/envoyconfig/builtingateway.go

```diff
@@ -0,0 +1,157 @@
+package envoyconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"slices"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/api/openapi/types"
+	api_common "github.com/kumahq/kuma/api/openapi/types/common"
+	meshtimeout "github.com/kumahq/kuma/pkg/plugins/policies/meshtimeout/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/test"
+	"github.com/kumahq/kuma/pkg/test/matchers"
+	"github.com/kumahq/kuma/pkg/test/resources/builders"
+	"github.com/kumahq/kuma/pkg/util/pointer"
+	. "github.com/kumahq/kuma/test/e2e_env/universal/gateway"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/envs/universal"
+)
+
+func BuiltinGateway() {
+	mesh := "envoyconfig-builtingateway"
+
+	const gatewayPort = 8080
+
+	gatewayAddressPort := func(appName string, port int) string {
+		ip := universal.Cluster.GetApp(appName).GetIP()
+		return net.JoinHostPort(ip, strconv.Itoa(port))
+	}
+
+	meshGateway := func() string {
+		return fmt.Sprintf(`
+type: MeshGateway
+name: gateway-proxy
+mesh: %s
+selectors:
+- match:
+    kuma.io/service: gateway-proxy
+conf:
+  listeners:
+  - port: %d
+    protocol: HTTP
+    hostname: example.kuma.io
+`, mesh, gatewayPort)
+	}
+
+	meshHTTPRoute := func() string {
+		return fmt.Sprintf(`
+type: MeshHTTPRoute
+name: gateway-proxy-route
+mesh: %s
+spec:
+  targetRef:
+    kind: MeshGateway
+    name: gateway-proxy
+  to:
+    - targetRef:
+        kind: Mesh
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: "/"
+          default:
+            backendRefs:
+              - kind: MeshService
+                name: echo-service
+                port: 80
+`, mesh)
+	}
+
+	BeforeAll(func() {
+		setup := NewClusterSetup().
+			Install(
+				Yaml(
+					builders.Mesh().
+						WithName(mesh).
+						WithoutInitialPolicies().
```

Comment:
don't we need mtls on mesh, like in sidecar tests, for testing MesTLS for example

Reply:
yes, added mtls